src/application.js

```diff
@@ -0,0 +1,50 @@
+import PuppyView from 'puppy-view';
+import CreateFormView from 'create-form';
+export default class ApplicationView {
+  constructor(el) {
+    this.el = el;
+
+    this.data = [];
+    this.el.querySelector('.top-nav__top-btn').addEventListener('click',()=>{
+    this.el.querySelector('.form-container').classList.toggle('form-container--active');
+  });
+
+  }
+  render(data) {
+    this.el.querySelector('.main-app').innerHTML = " ";
+    const components = this.data.map((item) => new PuppyView(item,this));
+
+    components.forEach((card) => {
+      this.el.appendChild(card.element);
```

Comment:
You are clearing main-app but then appending to the regular element. This is why you don't see the list clear on updates and deletes.